lighthouse-core/audits/seo/font-size.js

```diff
@@ -0,0 +1,286 @@
+  };
+}
+
+/**
+ * @param {string} baseURL
+ * @param {WebInspector.CSSStyleDeclaration} styleDeclaration
+ * @param {Node} node
+ * @returns {{source:!string, selector:string|object}}
+ */
+function findStyleRuleSource(baseURL, styleDeclaration, node) {
+  if (
+    !styleDeclaration ||
+    styleDeclaration.type === CSSStyleDeclaration.Type.Attributes ||
+    styleDeclaration.type === CSSStyleDeclaration.Type.Inline
+  ) {
+    return {
+      source: baseURL,
+      selector: nodeToTableNode(node),
+    };
+  }
+
+  if (styleDeclaration.parentRule &&
+    styleDeclaration.parentRule.origin === global.CSSAgent.StyleSheetOrigin.USER_AGENT) {
+    return {
+      selector: styleDeclaration.parentRule.selectors.map(item => item.text).join(', '),
+      source: 'User Agent Stylesheet',
+    };
+  }
+
+  if (styleDeclaration.type === CSSStyleDeclaration.Type.Regular && styleDeclaration.parentRule) {
+    const rule = styleDeclaration.parentRule;
+    const stylesheet = styleDeclaration.stylesheet;
+
+    if (stylesheet) {
+      let source;
+      const selector = rule.selectors.map(item => item.text).join(', ');
+
+      if (stylesheet.sourceURL) {
+        const url = parseURL(stylesheet.sourceURL, baseURL);
+        const range = styleDeclaration.range;
+        source = `${url.href}`;
+
+        if (range) {
+          // `stylesheet` can be either an external file (stylesheet.startLine will always be 0),
+          // or a <style> block (stylesheet.startLine will vary)
+          const absoluteStartLine = range.startLine + stylesheet.startLine + 1;
+          const absoluteStartColumn = range.startColumn + stylesheet.startColumn + 1;
+
+          source += `:${absoluteStartLine}:${absoluteStartColumn}`;
+        }
+      } else {
+        // dynamically injected to page
+        source = 'dynamic';
+      }
+
+      return {
+        selector,
+        source,
+      };
+    }
+  }
+
+  return {
+    source: 'Unknown',
+  };
+}
+
+/**
+ * @param {WebInspector.CSSStyleDeclaration} styleDeclaration
+ * @param {Node} node
+ * @return string
+ */
+function getFontArtifactId(styleDeclaration, node) {
+  if (styleDeclaration && styleDeclaration.type === CSSStyleDeclaration.Type.Regular) {
+    const startLine = styleDeclaration.range ? styleDeclaration.range.startLine : 0;
+    const startColumn = styleDeclaration.range ? styleDeclaration.range.startColumn : 0;
+    return `${styleDeclaration.styleSheetId}@${startLine}:${startColumn}`;
+  } else {
+    return `node_${node.nodeId}`;
+  }
+}
+
+class FontSize extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      name: 'font-size',
+      description: 'Document uses legible font sizes.',
+      failureDescription: 'Document doesn\'t use legible font sizes.',
+      helpText: 'Font sizes less than 16px are too small to be legible and require mobile ' +
+      'visitors to “pinch to zoom” in order to read. ' +
```

Comment:
it's a little funny to see this audit end up in passing even when there are failures ([example](https://user-images.githubusercontent.com/39191/33909742-7d2b3d6a-df41-11e7-8bff-8d15cd91734c.png)). I know this is the same as other audits, but i'm thinking we could start explaining our passing cutoff in the helpText here.

> Strive to have >75% of page text >=16px;

@vinamratasingal @kaycebasques  hows that sound?

Reply:
Updated!

<img width="794" alt="font-size-copy" src="https://user-images.githubusercontent.com/985504/33961835-c1dfe460-e04f-11e7-9b87-9128172832e1.png">